automata/base/automaton.py

```diff
@@ -124,29 +123,10 @@ def copy(self) -> Self:
         """Create a deep copy of the automaton."""
         return self.__class__(**self.input_parameters)
 
-    # Format the given value for string output via repr() or str(); this exists
```

Comment:
@eliotwrobson @khoda81 I originally wrote this function because calling `repr()` on an automaton instance was producing things like `frozenset({ ... })` and `frozendict({ ... })` in the output, but I wanted to display something friendlier like `{ ... }` and `{ "key": ... }` in the repr output. I thought this would be especially acceptable considering that native sets and dicts get type-cast to their frozen variants on instantiation anyway, so the repr output is still consistent with the code to create that instance.

I honestly don't know why this was removed, but it's my fault because I didn't say anything lol. I think it is probably worth keeping, though.

Reply:
@eliotwrobson @caleb531 Can we remove this function in the favour of `get_state_name(state)`? or remove `get_state_name(state)` in the favour of this? I think there is some duplicate logic here.

Reply:
If you can find a way to condense some of the logic that's fine, but the `get_state_name` function has a fundamentally different purpose than `_get_repr_friendly_value`, so I'm ok with having both if they're both short.

Reply:
@khoda81 So reviewing your code, it looks like `get_state_name` deals with a single state, but the purpose of `_get_repr_friendly_value` is more oriented towards handling `set` and `dict` structures.

But besides that, `get_state_name` does not look like it's designed for repr output, but rather, an aesthetically-pleasing display of the state used in an image of an automaton.

For example, one of the return values for `get_state_name` is `λ`, which makes sense when you are producing a rendered image of the machine. However, because `λ` is not considered the empty string when constructing an Automaton in Python code, `λ` would not make sense to use for repr output (because according to the Python docs, repr output should match the Python code to re-construct that object).

All that is to say, because the `get_state_name` is not designed for repr output and only operates on a single state, it has a fundamentally different purpose than `_get_repr_friendly_value`.

cc @eliotwrobson